drivers/input/input_ite_it8801_kbd.c

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (c) 2024 ITE Corporation. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ite_it8801_kbd
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/mfd/mfd_ite_it8801.h>
+#include <zephyr/input/input.h>
+#include <zephyr/input/input_kbd_matrix.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(input_ite_it8801_kbd, CONFIG_INPUT_LOG_LEVEL);
+
+struct it8801_mfd_input_altctrl_cfg {
+	/* GPIO control device structure */
+	const struct device *gpiocr;
+	/* GPIO control pin */
+	uint8_t pin;
+	/* GPIO function select */
+	uint8_t alt_func;
+};
+
+struct kbd_it8801_config {
+	struct input_kbd_matrix_common_config common;
+	/* IT8801 controller dev */
+	const struct device *mfd;
+	/* KSO alternate configuration */
+	const struct it8801_mfd_input_altctrl_cfg *altctrl;
+	/* I2C device for the MFD parent */
+	const struct i2c_dt_spec i2c_dev;
+	int mfdctrl_len;
+	uint8_t kso_mapping[DT_INST_PROP(0, col_size)];
+	/* Keyboard scan out mode control register */
+	uint8_t reg_ksomcr;
+	/* Keyboard scan in data register */
+	uint8_t reg_ksidr;
+	/* Keyboard scan in edge event register */
+	uint8_t reg_ksieer;
+	/* Keyboard scan in interrupt enable register */
+	uint8_t reg_ksiier;
+};
+
+struct kbd_it8801_data {
+	struct input_kbd_matrix_common_data common;
+	struct it8801_mfd_callback it8801_kbd_callback;
+};
+
+INPUT_KBD_STRUCT_CHECK(struct kbd_it8801_config, struct kbd_it8801_data);
+
+static void kbd_it8801_drive_column(const struct device *dev, int col)
+{
+	const struct kbd_it8801_config *config = dev->config;
+	int ret;
+	uint8_t kso_val;
+
+	/* Tri-state all outputs */
+	if (col == INPUT_KBD_MATRIX_COLUMN_DRIVE_NONE) {
+		/* KSO[22:11, 6:0] output high */
+		kso_val = IT8801_REG_MASK_KSOSDIC | IT8801_REG_MASK_AKSOSC;
+	}
+	/* Assert all outputs */
+	else if (col == INPUT_KBD_MATRIX_COLUMN_DRIVE_ALL) {
```

Comment:
should be `} else if {`, maybe move the comment after the condition and indent it?

Reply:
Done. Also make the comments above and below consistent.